pkg/controller/workload/job/job_webhook.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+)
+
+// log is for logging in this package.
+var jobLog = ctrl.Log.WithName("job-webhook")
+
+type JobWebhook struct {
+	manageJobsWithoutQueueName bool
+	client                     client.Client
+}
+
+// SetupWebhook configures the webhook for batchJob.
+func SetupWebhook(mgr ctrl.Manager, opts ...Option) error {
+	options := defaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	webhook := JobWebhook{
+		manageJobsWithoutQueueName: options.manageJobsWithoutQueueName,
+		client:                     mgr.GetClient(),
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&batchv1.Job{}).
+		WithDefaulter(&webhook).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-batch-v1-job,mutating=true,failurePolicy=fail,sideEffects=None,groups=batch,resources=jobs,verbs=create,versions=v1,name=mjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &JobWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *JobWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*batchv1.Job)
+	jobLog.V(5).Info("Applying defaults", "job", klog.KObj(job))
+
+	if queueName(job) == "" && w.manageJobsWithoutQueueName && !(*job.Spec.Suspend) {
+		job.Spec.Suspend = pointer.Bool(true)
+		return nil
+	}
+
+	if queueName(job) != "" && !(*job.Spec.Suspend) {
```

Comment:
wouldn't this cause `manageJobsWithoutQueueName` to be ignored?

Reply:
`manageJobsWithoutQueueName` only takes effect when it's true and queue name is empty, we already handled this above.

Reply:
manageJobsWithoutQueueName only takes effect when it's true and queue name is empty, we already handled this above.

Reply:
if manageJobWithoutQueueName is false, the above condition would be false, but this condition would be true.

Reply:
```
if manageJobWithoutQueueName == False && queueName(job) != "" && *job.Spec.Suspend == False {
    		job.Spec.Suspend = pointer.Bool(true)
}
```
Isn't this right?

Reply:
I think it might be?

But it's simpler to say something like this at the beginning:

```golang
if queueName(job) == "" && !manageJobWithoutQueue {
  return nil
}
```

Then you can apply suspend to any job

Reply:
Make sense